modules/ppcp-button/resources/js/modules/Renderer/CreditCardRenderer.js

```diff
@@ -194,6 +194,27 @@ class CreditCardRenderer {
             if (contingency !== 'NO_3D_SECURE') {
                 hostedFieldsData.contingencies = [contingency];
             }
+
+            if (this.defaultConfig.payer) {
+                hostedFieldsData.cardholderName = this.defaultConfig.payer.given_name + ' ' + this.defaultConfig.payer.surname;
+            }
+            if (!hostedFieldsData.cardholderName) {
+                const firstName = document.getElementById('billing_first_name') ? document.getElementById('billing_first_name').value
+                    : document.getElementById('shipping_first_name') ? document.getElementById('shipping_first_name').value
```

Comment:
This looks weird, why do we send the name for shipping address as cardholder name?

Reply:
@AlexP11223 

> This looks weird, why do we send the name from shipping address as cardholder name?

From where else could we get it when the customer is not logged in?

It appears that when no cardholder name is sent with the transaction, it causes displaying the name in PayPal as the DCC email (ex. example@dcc2.paypal.com or “CHASE VISA CARDHOLDER”).

https://developer.paypal.com/docs/business/checkout/advanced-card-payments/integrate

This PR ensures `cardholderName` is sent to hosted fields JSSDK otherwise it displays an error message and does not submit the hosted fields.

Reply:
Another solution could be to add a new "Card Holder" field to hosted fields:
![02](https://user-images.githubusercontent.com/456223/144039172-f29eb6bd-b9fa-48f9-ab3d-152da52f6dc6.png)

Reply:
Shipping address may not be related to payment. 

Why not just show error if no name in the billing address?

Reply:
> Shipping address may not be related to payment.

yup makes totally sense, I´ve just removed shipping as fallback, thanks.